tools/integration_tests/test_storage.py

```diff
@@ -0,0 +1,2077 @@
+        self.assertTrue("X-Weave-Next-Offset" not in res.headers)
+
+        res = self.app.get(
+            query_url + '&limit=10000&offset=' + next_offset)
+        self.assertEquals(res.json, all_items[5:])
+
+        # "offset" once more, this time with no explicit ordering
+        query_url = self.root + '/storage/xxx_col2?'
+        res = self.app.get(query_url)
+        all_items = res.json
+        self.assertEquals(len(all_items), 10)
+
+        res = self.app.get(query_url + '&limit=2')
+        self.assertEquals(res.json, all_items[:2])
+
+        next_offset = res.headers["X-Weave-Next-Offset"]
+        res = self.app.get(query_url + '&limit=3&offset=' + next_offset)
+        self.assertEquals(res.json, all_items[2:5])
+
+        next_offset = res.headers["X-Weave-Next-Offset"]
+        res = self.app.get(query_url + '&offset=' + next_offset)
+        self.assertEquals(res.json, all_items[5:])
+        self.assertTrue("X-Weave-Next-Offset" not in res.headers)
+
+        res = self.app.get(
+            query_url + '&limit=10000&offset=' + next_offset)
+
+        # "sort"
+        #   'newest': Orders by timestamp number (newest first)
+        #   'oldest': Orders by timestamp number (oldest first)
+        #   'index':  Orders by the sortindex descending (highest weight first)
+        self.retry_delete(self.root + '/storage/xxx_col2')
+
+        for index, sortindex in (('00', -1), ('01', 34), ('02', 12)):
+            bso = {'id': index, 'payload': 'x', 'sortindex': sortindex}
+            self.retry_post_json(self.root + '/storage/xxx_col2', [bso])
+
+        res = self.app.get(self.root + '/storage/xxx_col2?sort=newest')
+        res = res.json
+        self.assertEquals(res, ['02', '01', '00'])
+
+        res = self.app.get(self.root + '/storage/xxx_col2?sort=oldest')
+        res = res.json
+        self.assertEquals(res, ['00', '01', '02'])
+
+        res = self.app.get(self.root + '/storage/xxx_col2?sort=index')
+        res = res.json
+        self.assertEquals(res, ['01', '02', '00'])
+
+    def test_alternative_formats(self):
+        bsos = [{"id": str(i).zfill(2), "payload": "xxx"} for i in range(5)]
+        self.retry_post_json(self.root + "/storage/xxx_col2", bsos)
+
+        # application/json
+        res = self.app.get(self.root + '/storage/xxx_col2',
+                           headers=[('Accept', 'application/json')])
+        self.assertEquals(res.content_type.split(";")[0], 'application/json')
+
+        res = res.json
+        res.sort()
+        self.assertEquals(res, ['00', '01', '02', '03', '04'])
+
+        # application/newlines
+        res = self.app.get(self.root + '/storage/xxx_col2',
+                           headers=[('Accept', 'application/newlines')])
+        self.assertEquals(res.content_type, 'application/newlines')
+
+        self.assertTrue(res.body.endswith(b'\n'))
+        res = [json_loads(line) for line in res.body.decode('utf-8').strip().split('\n')]
```

Comment:
heh, yeah, we should run flake8 on this, just so we're consistent about things...

Reply:
and maybe also add to CI?

Reply:
It's in ci already @tublitzed.

Reply:
python linting is? great! I must have missed it then. @fzzzy can you point me to where we're doing that? I'm probably overlooking it but not seeing anything [here](https://github.com/mozilla-services/syncstorage-rs/blob/master/.circleci/config.yml)

Reply:
Ok, so apparently it's not in syncstorage-rs! I guess this is the first time we have any python files. I'll add it. (We relied on it being enabled in server-syncstorage)

Reply:
The tool that we tend to use for this is Flake8. It's possible to add it into the CI process, but considering the general complexity that we already have, I don't really think we need to compound that. Also, we shouldn't be changing this file frequently, so running flake8 on this file every build just burns a not insignificant amount of CPU. Plus, flake8 isn't like `cargo fmt`, you have to manually edit the file to resolve issues it may find. 

We can run it manually if we ever touch these files and generally check for visual clues that it might be needed. (e.g. lines longer than 80 characters, some spacing issues, etc.)

Reply:
> python linting is? great! I must have missed it then. @fzzzy can you point me to where we're doing that? I'm probably overlooking it but not seeing anything [here](https://github.com/mozilla-services/syncstorage-rs/blob/master/.circleci/config.yml)

Ah, right, it wasn't enabled for syncstorage-rs yet because we haven't yet had any python files. I added a new bug to track adding flake8 coverage to the new python3 test suite.